lib/docker/image.js

```diff
@@ -93,7 +98,13 @@ class Image {
   }
 
   async build() {
-    const stream = execa('docker', this.build_cmd)
+    const env = {}
+    // Setting a stage as target requires buildkit to be enabled
+    if (this.opts.target !== null) {
+      env.DOCKER_BUILDKIT = 1
```

Comment:
There are other ways to enable this. I think it may be better to just specify in the documentation that build kit must be enabled for this feature. Its not really the concern of this plugin to do that.